tests/frontend/test_engine.py

```diff
@@ -156,3 +168,40 @@ def inspect():
         eng.reset()
         eng.run([p1, p2])
         assert inspect() == expected2
+
+
+class TestMultipleShotsErrors:
+    """Test if errors are raised correctly when using multiple shots."""
+
+    @pytest.mark.parametrize("meng", batch_engines)
+    def test_batching_error(self, meng, prog):
+        """Check that correct error is raised with batching and shots > 1."""
+        with pytest.raises(NotImplementedError) as excinfo:
```

Comment:
Note that pytest has a 'match' option:
```suggestion
        with pytest.raises(NotImplementedError, match="Batching"):
```

Reply:
Oh, that's nicer. I didn't know that. :smile: Thanks!